### PR TITLE
Sweets/claim

### PR DESCRIPTION
**Pull request by sweetmantech**
- new function: `claimPoster` - claims a Poster. ✨ 
- Poster struct - new attribute - `kind`. ✨ 
- new Enum - `PosterKind` - `[IMAGE, VIDEO]`. ✨ 
- new modifier - `mustNotBeTokenOwner` - ensures only people who don't own tokens can mint a poster. (EXPLOITABLE) ✨

### CODE REVIEW

**Review by waynehoover — APPROVED** [review]

File: contracts/PostersSeason2ByMintSongs.sol [hunk 1]
```diff
@@ -86,12 +109,33 @@ contract PosterFactory is ERC1155OnChainMetadata, IERC2981Upgradeable {
...
-            _royaltyRecipient
+            _royaltyRecipient,
+            _media_type
         );
         poster[id] = newPoster;
         return id;
     }
 
+    /**
+     * @dev airdrop 1 free poster
```
Comment (blocktator) [inline contracts/PostersSeason2ByMintSongs.sol]: is this considered airdropping given they are collected directly against the contract?

File: contracts/PostersSeason2ByMintSongs.sol [hunk 2]
```diff
@@ -86,12 +109,33 @@ contract PosterFactory is ERC1155OnChainMetadata, IERC2981Upgradeable {
...
+        external
+        mustNotBeTokenOwner(_id, _msgSender())
+    {
+        require(
+            poster[_id].count > 0,
+            "MintSongs1155: claim non-existing poster"
+        );
+        require(
+            poster[_id].maxSupply >= (poster[_id].count + 1),
+            "cannot mint more than max supply"
```
Comment (blocktator) [inline contracts/PostersSeason2ByMintSongs.sol]:
```suggestion
            "cannot claim more than the max supply"
```

**Review by blocktator — APPROVED** [review]
just a minor suggestion, but good to go

File: contracts/PostersSeason2ByMintSongs.sol (hunk 1 above)
Comment (sweetmantech) [reply contracts/PostersSeason2ByMintSongs.sol]: whoops good catch, now that we've removed the `_to` param it's not an airdrop.

Comment (sweetmantech) [reply contracts/PostersSeason2ByMintSongs.sol]: updated ✅ 

Screen Shot 2022-07-14 at 1 22 20 PM

**Pull request merged**